src/dmd/root/filename.d

```diff
@@ -631,71 +631,75 @@ nothrow:
         }
     }
 
+    /**
+       Ensure that the provided path exists
+
+       Accepts a path to either a file or a directory.
+       In the former case, the `basepath` (path to the containing directory)
+       will be checked for existence, and created if it does not exists.
+       In the later case, the directory pointed to will be checked for existence
+       and created if needed.
+
+       Params:
+         path = a path to a file or a directory
+
+       Returns:
+         `true` if the directory exists / was successfully created,
+         `false` otherwise
+     */
     extern (C++) static bool ensurePathExists(const(char)* path)
     {
         //printf("FileName::ensurePathExists(%s)\n", path ? path : "");
-        if (path && *path)
+        if (!path || !(*path))
+            return true;
+        if (exists(path))
+            return true;
+
+        // We were provided with a file name
+        // We need to call ourselves recursively to ensure parent dir exist
+        const(char)* p = FileName.path(path);
+        if (*p)
         {
-            if (!exists(path))
+            version (Windows)
             {
-                const(char)* p = FileName.path(path);
-                if (*p)
+                size_t len = strlen(path);
```

Comment:
I though `strlen` was supposed to die.  Wile you're at it, can't you change the interface to accept a `string` or other `char[]`.

Reply:
I have a WIP which converts everything in filename to use slices, which is how I came across this code. But it's currently blocked on my JSON pr so I decided to submit this refactoring first.